docs/tutorials/Remark.adoc

```diff
@@ -0,0 +1,360 @@
+provided for the function to understand what it does.
+
+[source, java]
+.ArgumentTokenizer.java
+----
+/**
+ * Tokenizes an arguments string and returns an {@code ArgumentMultimap} object that maps prefixes to their
+ * respective argument values. Only the given prefixes will be recognized in the arguments string.
+ *
+ * @param argsString Arguments string of the form: {@code preamble <prefix>value <prefix>value ...}
+ * @param prefixes   Prefixes to tokenize the arguments string with
+ * @return           ArgumentMultimap object that maps prefixes to their arguments
+ */
+----
+
+We can tell `ArgumentTokenizer#tokenize()` to look out for our new prefix `r/` and it will return us an instance of
+`ArgumentMultimap`. Now let's find out what we need to do in order to obtain the Index and String that we need.
+Let's look through ArgumentMultimap :
+
+[source, java]
+.ArgumentMultimap.java
+----
+/**
+ * Returns the last value of {@code prefix}.
+ */
+public Optional<String> getValue(Prefix prefix) {
+    List<String> values = getAllValues(prefix);
+    return values.isEmpty() ? Optional.empty() : Optional.of(values.get(values.size() - 1));
+}
+----
+
+This appears to be what we need to get a String of the remark. But what about the Index? Taking a quick peek at
+existing an `Command`...
+
+[source, java]
+.DeleteCommandParser.java
+----
+Index index = ParserUtil.parseIndex(args);
+return new DeleteCommand(index);
+----
+
+There appears to be another utility class that obtains an `Index` from the input provided by the user.
+
+Now that we have the know-how to extract the data that we need from the user's input, we can instantiate
+`RemarkCommand` by passing the data through its constructor.
+
+[source, java]
+.RemarkCommandParser.java
+----
+public RemarkCommand parse(String args) throws ParseException {
+    requireNonNull(args);
+    ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_REMARK);
+
+    Index index;
+    try {
+        index = ParserUtil.parseIndex(argMultimap.getPreamble());
+    } catch (IllegalValueException ive) {
+        throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemarkCommand.MESSAGE_USAGE), ive);
+    }
+
+    String remark = argMultimap.getValue(PREFIX_REMARK).orElse("");
+
+    return new RemarkCommand(index, remark);
+}
+----
+
+NOTE: Don't forget to update `AddressBookParser` to use our new `RemarkCommandParser`!
+
+If you are stuck, check out the sample
+link:++https://github.com/j-lum/addressbook-level35/commit/c0f8419be4fb59bffafcd73026769459e5cf5c5e#diff
+-fc19ecee89c3732a62fbc8c840250508++[here].
+
+== Adding to the model
+
+Now that we have all the information that we need, let's lay the groundwork for some _persistent_ changes. We acheive
+ that by working with the `Person` model. Each field in a Person is encapsulated. This protects them from illegal
+ access and allows us to tightly couple logic specific to the field.
```

Comment:
>Each field in a Person is encapsulated. This protects them from illegal  access and allows us to tightly couple logic specific to the field.

-> Something like this?

>Each field in a Person is implemented as a separate class (e.g., a `Name` object represents the person's name). That means we should add a `Remark` class so that we can use a `Remark` object to represent a remark given to a person.